organisaatio-ui/src/components/Modaalit/ToimipisteenNimenmuutos/ToimipisteenNimenmuutosModaali.tsx

```diff
@@ -3,38 +3,108 @@ import TNBody from './TNBody';
 import * as React from 'react';
 import { useForm } from 'react-hook-form';
 import { joiResolver } from '@hookform/resolvers/joi';
-import { Nimi } from '../../../types/types';
+import { LocalDate, NimenmuutosLomake, OrganisaationNimetNimi } from '../../../types/types';
 import ToimipisteenNimenmuutosModaaliSchema from '../../../ValidationSchemas/ToimipisteenNimenmuutosModaaliSchema';
 import Header from '../Header/Header';
 import Footer from '../Confirmation/Footer';
+import { useState } from 'react';
+import { createOrganisaatioNimi, updateOrganisaatioNimi } from '../../../api/organisaatio';
+import { MUUTOSTYYPPI_CREATE, MUUTOSTYYPPI_EDIT } from './constants';
 
 type ModaaliProps = {
-    nimi: Nimi;
-    closeNimenmuutosModaali: () => void;
-    handleNimiTallennus: (Nimi) => void;
+    oid: string;
+    currentNimi?: OrganisaationNimetNimi & { disabled: boolean | undefined };
+    closeNimenmuutosModaali: (nimiIsUpdated: boolean) => void;
 };
+
 export default function ToimipisteenNimenmuutosModaali(props: ModaaliProps) {
+    const { currentNimi, oid } = props;
+    const [isLoading, setIsLoading] = useState<boolean>(false);
     const {
         reset,
         getValues,
         register,
         formState: { errors: validationErrors },
         handleSubmit,
-    } = useForm<Nimi>({ defaultValues: props.nimi, resolver: joiResolver(ToimipisteenNimenmuutosModaaliSchema) });
+        control: formControl,
+        watch,
+    } = useForm<NimenmuutosLomake>({
+        defaultValues: {
+            nimi: { fi: '', sv: '', en: '' },
+            alkuPvm: new Date().toISOString().split('T')[0] as LocalDate,
+            muutostyyppi: MUUTOSTYYPPI_CREATE,
+            oid,
+            editDisabled: currentNimi?.disabled,
+        },
+        resolver: joiResolver(ToimipisteenNimenmuutosModaaliSchema),
+    });
+
+    React.useEffect(() => {
+        const subscription = watch((value, { name }) => {
+            if (name === 'muutostyyppi') {
+                const { muutostyyppi, editDisabled } = value;
+                switch (muutostyyppi) {
+                    case MUUTOSTYYPPI_CREATE:
+                        reset({
+                            nimi: { fi: '', sv: '', en: '' },
+                            alkuPvm: new Date().toISOString().split('T')[0] as LocalDate,
```

Comment:
maybe use moment since we have it already?

Reply:
ill change